client/src/pages/GitHubRedirect.js

```diff
@@ -0,0 +1,18 @@
+import React, { useContext, useEffect } from "react";
+import { useLocation, Redirect } from "react-router-dom";
+import { AuthContext } from "../contexts/AuthContext";
+import Page from "../common/Page";
+
+export default function GitHubRedirect() {
+  const { loginWithCode, user } = useContext(AuthContext);
+  const search = new URLSearchParams(useLocation().search);
+  const code = search.get("code");
+
+  useEffect(() => {
+    if (code) {
+      loginWithCode(code);
+    }
+  }, [code, loginWithCode]);
```

Comment:
Right now, the `useEffect` is called after every render because `loginWithCode` is recreated on every render.
This will be fixed by `useCallback` of my previous comment.